tests/functional/test_config_groups.py

```diff
@@ -362,69 +364,68 @@ class TestChangeGroupsConfig:
+        "structure": [{"code": 3, "country": "Test1"}, {"code": 4, "country": "Test2"}],
+        "map": {"age": "20", "name": "Chloe", "sex": "f"},
+        "password": "123",
+        "file": "file content test",
+    }
+
+    GROUP_KEYS_TO_CHANGE = {
+        "float": True,
+        "boolean": True,
+        "integer": True,
+        "password": True,
+        "string": True,
+        "list": True,
+        "file": True,
+        "option": True,
+        "text": True,
+        "group": {"port": True, "transport_port": True},
+        "structure": True,
+        "map": True,
+    }
+
+    CLUSTER_HOSTS_VARIANTS = [
+        "all",
+        "CLUSTER",
+        "test_service_1",
+        "test_service_1.first",
+        "test_service_1.second",
+        FIRST_HOST,
+        SECOND_HOST,
     ]
 
-    def _add_values_to_group_config_template(self, params: list = PARAMS_TO_CHANGE) -> dict:
-        (
-            float_value,
-            boolean_value,
-            integer_value,
-            string_value,
-            list_value,
-            option,
-            text,
-            group,
-            structure,
-            map_value,
-            password,
-            file,
-        ) = params
-        return {
-            "attr": {
-                "group": {"active": True},
-                "group_keys": {
-                    "group": {"port": True, "transport_port": True},
-                    "float": True,
-                    "boolean": True,
-                    "integer": True,
-                    "password": True,
-                    "string": True,
-                    "list": True,
-                    "file": True,
-                    "option": True,
-                    "text": True,
-                    "structure": True,
-                    "map": True,
-                },
-            },
-            "config": {
-                "float": float_value,
-                "boolean": boolean_value,
-                "integer": integer_value,
-                "password": password,
-                "string": string_value,
-                "list": list_value,
-                "file": file,
-                "option": option,
-                "text": text,
-                "group": group,
-                "structure": structure,
-                "map": map_value,
-            },
-        }
+    def _add_values_to_group_config_template(
+        self, custom_group_keys: dict = {}, group_keys: dict = {}, config_attr: dict = PARAMS_TO_CHANGE
+    ) -> dict:
+        group_config_template = {"attr": {}, "config": {}}
+        if custom_group_keys:
+            group_config_template["attr"]["custom_group_keys"] = {}
+            if "group" in custom_group_keys.keys():
+                group_value = custom_group_keys.pop("group")
+                group_config_template["attr"]["custom_group_keys"]["group"] = {
+                    "port": group_value,
+                    "transport_port": group_value,
+                }
```

Comment:
Why do you want to hardcode keys here instead of always passing group as `{"port": x, "transport_port": x}`?  Aren't there any cases, when you may want to pass something else in the "group" key? If it's required to be that way please consider mentioning this behavior in docstring or comment.

Reply:
changed